gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -317,6 +394,30 @@ public void run() {
       }
     }
 
+    private void cancelDag(String dagToCancel) throws ExecutionException, InterruptedException {
+      log.info("Cancel flow with DagId {}", dagToCancel);
+      if (this.dagToJobs.containsKey(dagToCancel)) {
+        List<DagNode<JobExecutionPlan>> dagNodesToCancel = this.dagToJobs.get(dagToCancel);
+        log.info("Found {} DagNodes to cancel.", dagNodesToCancel.size());
+        for (DagNode<JobExecutionPlan> dagNodeToCancel : dagNodesToCancel) {
+          Properties props = new Properties();
+          if (dagNodeToCancel.getValue().getJobFuture().isPresent()) {
+            Future future = dagNodeToCancel.getValue().getJobFuture().get();
+            if (future instanceof CompletableFuture &&
```

Comment:
Assuming that the if {..} block will be moved inside AzkabanSpecProducer#deleteSpec() implementation? If so, add a TODO there.

Reply:
I am not planning to move it to AzkabanSpecProducer; because we can extract azkabanExecId here only. Otherwise, we will have to pass the future object (which has the azkabanExecId) to AzkabanSpecProducer#deleteSpec()